src/mon/OSDMonitor.cc

```diff
@@ -14629,6 +14629,15 @@ int OSDMonitor::_prepare_remove_pool(
       }
     }
   }
+  // remove any pg_upmap_primary mapping for this pool
+  for (auto& p : osdmap.pg_upmap_primaries) {
+    if (p.first.pool() == pool) {
+      dout(10) << __func__ << " " << pool
+               << " removing obsolete pg_upmap_primaries " << p.first
+               << dendl;
+      pending_inc.old_pg_upmap_primary.insert(p.first);
+    }
+  }
```

Comment:
Thanks for this @mohit84! Can you also include removal of `pending_inc.new_pg_upmap_primary`? Similar to the above section which removes `new_pg_upmap_items`.

Reply:
done

Reply:
Thank you!